bigquery_etl/query_scheduling/templates/airflow_dag.j2

```diff
@@ -542,4 +532,18 @@ with DAG('{{ name }}', default_args=default_args{%+ if schedule_interval != None
 
     {% endfor -%}
     {% endif -%}
+
+    {% if task.depends_on_fivetran != None -%}
+    {% for fivetran_task in task.depends_on_fivetran -%}
+    {% for fivetran_dependency in fivetran_task.depends_on | sort(attribute='task_id') -%}
+    {% if fivetran_dependency.dag_name == name and fivetran_dependency.get_execution_delta(schedule_interval) in [none, '0h', '0m', '0s'] -%}
+        {% if fivetran_dependency.task_id != fivetran_task.task_name -%}
+        {{ fivetran_task.task_id }}_sync_start.set_upstream({{ fivetran_dependency.task_id }})
+        {% endif -%}
+    {% else -%}
+        {{ fivetran_task.task_id }}_sync_start.set_upstream(wait_for_{{ fivetran_dependency.task_id | replace('-', '_') }})
+    {% endif -%}
```

Comment:
If multiple ETLs depend on the same Fivetran task the same `set_upstream()` calls could get output multiple times (assuming each ETL's `depends_on_fivetran` metadata specified the nested `depends_on` metadata, which they ought to for consistency).  However, I don't think that would cause any actual problem, the DAG code would just be slightly duplicative.

Reply:
I've actually been thinking it'd be better to not have the Fivetran tasks be solely defined in ETL metadata, because when multiple ETLs depend on the same Fivetran task you have to be careful to keep that Fivetran dependency metadata exactly the same in all cases to avoid issues.

Maybe the Fivetran tasks could be defined at the DAG level in `dags.yaml`, and then the ETL dependencies on those Fivetran tasks could be specified via the normal `depends_on` metadata?